ideasbox/library/models.py

```diff
@@ -71,12 +71,21 @@ def index_strings(self):
 class BookSpecimen(TimeStampedModel):
 
     book = models.ForeignKey(Book, related_name='specimens')
-    serial = models.CharField(_('serial'), max_length=40, unique=True)
+    serial = models.CharField(_('serial'), max_length=40, unique=True, blank=True)
     location = models.CharField(_('location'), max_length=300, blank=True)
     remarks = models.TextField(_('remarks'), blank=True)
+    spfile = models.FileField(_('file'), upload_to='library/file', blank=True)
+
+    def is_digital(self):
+        return (u'{0}'.format(self.spfile) != '')
```

Comment:
You can just check for the [attribute itself](https://docs.djangoproject.com/en/1.8/ref/forms/fields/#filefield), something like `return self.spfile is not None`